File: sources/nest-backend/src/job/dto/update-job.dto.ts
import { ApiProperty, PartialType } from '@nestjs/swagger';
import { CreateJobDto, RecurringType } from './create-job.dto';
import { IsBoolean, IsDate, IsEnum, IsNumber, IsString } from 'class-validator';
import { TransactionType } from 'src/transaction/transaction.service';

export class UpdateJobDto extends PartialType(CreateJobDto) {
  @ApiProperty({ example: '2024-01-01' })
  @IsDate()
  schedule_date: Date;

  @ApiProperty({ example: 100 })
  @IsNumber()
  amount: number;

  @ApiProperty({ example: 'DAILY' })
  @IsString()
  @IsEnum(RecurringType)
  recurring_type: RecurringType;

  @ApiProperty({ example: 'Sample job name' })
  @IsString()
  job_name: string;

  @ApiProperty({ example: '2024-01-01' })
  @IsDate()
  next_run_date: Date;

  @ApiProperty({ example: 'true' })
  @IsBoolean()
  status: boolean;

  @ApiProperty({ example: '1' })
  @IsNumber()
  tag_id: number;

  @ApiProperty({ example: '1' })
  @IsNumber()
  user_id: number;

  @ApiProperty({ example: '1' })
  @IsNumber()
  account_id: number;

  @ApiProperty({ example: 'INCOME' })
  @IsEnum(TransactionType)
  transaction_type: TransactionType;

  @ApiProperty({ example: 'description' })
  @IsString()
  description: string;
}

File: sources/win-ui-frontend/Fin-Manager-v2/DTO/UpdateJobDto.cs
using System.Text.Json.Serialization;

namespace Fin_Manager_v2.DTO;

public class UpdateJobDto
{

    [JsonPropertyName("job_name")]
    public string JobName { get; set; } = string.Empty;

    [JsonPropertyName("job_description")]
    public string Description { get; set; } = string.Empty;

    [JsonPropertyName("tag_id")]
    public int TagId { get; set; }

    [JsonPropertyName("account_id")]
    public int AccountId { get; set; }

    [JsonPropertyName("user_id")]
    public int UserId { get; set; }

    [JsonPropertyName("schedule_date")]
    public string ScheduleDate { get; set; } = string.Empty;


    [JsonPropertyName("amount")]
    public decimal Amount { get; set; }

    [JsonPropertyName("recurring_type")]
    public string RecurringType { get; set; } = string.Empty;

    [JsonPropertyName("transaction_type")]
    public string TransactionType { get; set; } = string.Empty;
} 